### PR TITLE
SCMOD-3478: Update processing API to store global & tenants in database

### PR DESCRIPTION
**Pull request by mmaBel**
https://jira.autonomy.com/browse/SCMOD-3478

### CODE REVIEW

**Comment by mmaBel** [comment]
This shouldn't be merged until Anthony's [SCMOD-3111](https://github.com/CAFDataProcessing/data-processing-service/pull/33) & Rado's SCMOD-3477 changes are approved as it can't be tested by QA without them.

Also for testing, the deploy branches need updated with the configuration to talk to the database. The following pull requests can be merged before this one to support this;
https://github.houston.softwaregrp.net/caf/data-processing-service-internal-deploy/pull/68/files
https://github.com/CAFDataProcessing/data-processing-service/pull/33/files
https://github.com/CAFDataProcessing/policy-server/pull/11/files

File: processing-service-core/api/models/db/tables/tenantConfig.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        allowNull: false,
+        field: "tenant_id",
+        primaryKey: true,
+        type: Sequelize.STRING(40)
+    },
+    key: {
+        allowNull: false,
+        field: "key",
+        primaryKey: true,
+        // TODO how to represent the foreign key constraint?
```
Comment (dermot-hardy) [inline processing-service-core/api/models/db/tables/tenantConfig.js]: Is there anything more to do here?

**Review by dermot-hardy — APPROVED** [review]

File: processing-service-core/api/models/db/tables/tenantConfig.js (hunk 1 above)
Comment (mmaBel) [reply processing-service-core/api/models/db/tables/tenantConfig.js]: Updated to add in reference to the global config 'key' column. Couldn't find an example of doing so without creating an instance of the other model which is why I'd originally added that comment but source code indicates that table name can be used here.
http://docs.sequelizejs.com/manual/tutorial/models-definition.html#configuration
- describes setting up the association

The foreign key constraint was already being enforced as it is set up by the DB installer. This is just about having it written down in the definition here.

**Review by Anthony-McGreevy — APPROVED** [review]

**Comment by mmaBel** [comment]
Review comment for tenant config contract pull request [here](https://github.com/CAFDataProcessing/data-processing-service/pull/33) says that 404 should not occur on getTenantConfigs if no items are found. Changed this behavior in DB store code.

**Comment by mmaBel** [comment]
@Anthony-McGreevy please take another look at this following the commits to the tests and the tenant model.

**Review by Anthony-McGreevy — APPROVED** [review]

**Pull request merged**